### PR TITLE
Remove inline deployers and add additional deployer configuration

### LINKED ISSUE #140: Remove inline deployers from landscaper production binary
**How to categorize this issue?**

/area operations
/kind cleanup
/priority 4

**What would you like to be added**:

Currently our maintained deployers (helm, manifest, container) are also vendored in the landscaper.
This makes is really easy to start with the landscaper development.

But we should only include these deployers in a specific dev image and remove them from our production landscaper image.

### PR DESCRIPTION
**Pull request by schrodit**
**How to categorize this PR?**

/area deployers
/kind task
/priority 3

**What this PR does / why we need it**:
Follow up of https://github.com/gardener/landscaper/pull/185

This PR completely removes the inline deployer and replaces them with the DLM based deployment.
It is now also possible to define additional deployers in the `deployersConfig` so that external deployers can be managed within one landscaper deployment.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/landscaper/issues/140

**Special notes for your reviewer**:

As the deployer options and deployment was getting too big and to complex for the just the command package.
I moved them to a dedicated package in `pkg/deployermanagement` to show that they are not a workaround anymore but are part of the stable landscaper api.
With this also the warning in the docs about the internal deployers have been removed as the DLM based deployer are production ready.

**Release note**:

```breaking operator
Inline deployer have been completely removed and are replaced with the DLM based deployment.
It is now also possible to define additional deployers in the `deployersConfig` so that external deployers can be managed within one landscaper deployment.
```

### CODE REVIEW

File: pkg/deployermanagement/config/deployment.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+func (o *Options) deployInternalDeployer(ctx context.Context, log logr.Logger, deployerName string, kubeClient client.Client, apply DeployerApplyFunc) error {
+	log.Info("Enable Deployer", "name", deployerName)
+
+	deployerConfig, err := o.GetDeployerConfigForDeployer(deployerName)
+	if err != nil {
+		if err != NoConfigError {
+			return fmt.Errorf("unable to get Deployer Config: %w", err)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (schrodit) [reply pkg/deployermanagement/config/deployment.go:108]: good catch. looks like a remnant of somme refactorings.
In fact we can completely ignore the error as the function always returns a config (types differ).

File: docs/gettingstarted/install-landscaper-controller.md [hunk 2]
```diff
@@ -86,12 +86,64 @@ The value to provide to `<docker auth>` must be a Docker auth config as plain JS
...
+```yaml
+landscaper:
+  deployers: [my-custom-deployer] # the external deployer name MUST be set.
+  deployersConfig:
+    # match the deployer name
+    my-custom-deployer: 
+      apiVersion: landscaper.gardener.cloud/v1alpha1
+      kind: DeployerRegistration
+      metadata:
+        name: my-deployer # defaulted to "my-custom-deployer"
```
[the remark this answers is not in the document]
Comment (schrodit) [reply docs/gettingstarted/install-landscaper-controller.md:130]: yes that name can diiffer.
The idea is that by default the deployer registration will be defaulted to the name of the key.
But if the actual registration should have a different name than the key in the helm chart it is possible to overwrite.

So would think of it more as a optional configuration

**Review by achimweigel — APPROVED** [review]
/lgtm

**Pull request merged**